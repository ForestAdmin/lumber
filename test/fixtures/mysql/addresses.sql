CREATE TABLE addresses (
  id VARCHAR(36) NOT NULL PRIMARY KEY,
  customer_id INT UNSIGNED NULL,
  city VARCHAR(30) NOT NULL UNIQUE,
  CONSTRAINT fk_customer_id FOREIGN KEY (customer_id) REFERENCES customers (id)
);
