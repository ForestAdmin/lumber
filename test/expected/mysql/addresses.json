{
  "fields": [
    {
      "name": "id",
      "nameColumn": "id",
      "type": "STRING",
      "primaryKey": true,
      "defaultValue": null
    },
    {
      "name": "city",
      "nameColumn": "city",
      "type": "STRING",
      "primaryKey": false,
      "defaultValue": null
    }
  ],
  "references": [
    {
      "as": "customer",
      "foreignKey": "customer_id",
      "foreignKeyName": "customerId",
      "ref": "customers"
    }
  ],
  "primaryKeys": [
    "id"
  ],
  "options": {
    "underscored": false,
    "timestamps": false,
    "hasIdColumn": true,
    "hasPrimaryKeys": true
  }
}
